### DISCUSSION TITLE
How to set target value in eigensolver?

### DISCUSSION THREAD

**Discussion by rainiscold** [idaholab/moose#28555] [Q&A Modules: General] [answered]
Is there an interface in Moose to set the target value in eigen solver? 
I know it is possible in SLEPc, I couldn't figure out how to set it in Moose.

**Comment by GiudGiud** [comment]
Hello

Do you mean setting a target value then adjusting parameters to make the solution be this target value?
Or do you mean selecting an eigenmode by selecting a known eigenvalue?

**Comment by maxnezdyur** [comment, marked as answer]
I would look at this example `modules/solid_mechanics/test/tests/modal_analysis/modal.i`.
We can pass any petsc/slepc options using the interface
```
petsc_options_iname = '-st_type -eps_target -st_pc_type -st_pc_factor_mat_solver_type'
petsc_options_value = 'sinvert 0 lu mumps'
```

**Reply by rainiscold** [reply]
Thank you for the info. I didn't realize the solution is right there.
